src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs

```diff
@@ -166,6 +167,24 @@ private static string NormalizedBlobPath(string blobPath)
             return string.IsNullOrEmpty(blobPath) ? string.Empty : $"/{blobPath.TrimStart('/')}";
         }
 
+        /// <inheritdoc />
+        public override string GetInternalTesTaskBlobUrlWithoutSasToken(TesTask task, string blobPath)
+        {
+            var normalizedBlobPath = NormalizedBlobPath(blobPath);
+            var blobPathWithPrefix = $"{task.Id}{normalizedBlobPath}";
+            if (task.Resources?.ContainsBackendParameterValue(TesResources.SupportedBackendParameters
+                    .internal_path_prefix) == true)
+            {
+                blobPathWithPrefix =
+                    $"{task.Resources.GetBackendParameterValue(TesResources.SupportedBackendParameters.internal_path_prefix).Trim('/')}{normalizedBlobPath}";
+            }
+
+            //passing the resulting string through the builder to ensure that the path is properly encoded and valid
+            var builder = new BlobUriBuilder(new Uri($"https://{defaultStorageAccountName}.blob.core.windows.net/{blobPathWithPrefix.TrimStart('/')}"));
```

Comment:
is this missing the container name?

Reply:
Yes, good catch. This method is not used. I will clean this up.